shm/Map.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <random>
+#include <vector>
+#include "Island.hpp"
+
+constexpr auto defaultIslandsNumber = 10u;
```

Comment:
move to class. If this value is needed only in .cpp file then move to the anonymous namespace in .cpp file.